## Reading in files (csv)
- Packages|module|libray: a bunch of functions/methods bundled up as a chunk to conveniently address a collection of needs(data visualization: "seaborn", "matplotlib.pyplot", data manipulation: "pandas", "numpy") 
    - pandas: for reading and manipulating data

In [4]:
# viewing and changing working directory
import os
os.getcwd()

'C:\\Non-Default aka Real Shit\\College is a hot mess\\Spring 2019\\430\\Markdown'

In [ ]:
os.chdir("C:\\Non-Default aka Real Shit\\College is a hot mess\\Spring 2019\\430\\Markdown")
os.getcwd()

In [1]:
import pandas as pd

In [2]:
avocado_df = pd.read_csv("avocado.csv")

In [3]:
avocado_df.head()

Unnamed: 0        Date  AveragePrice  Total Volume     4046       4225  \
0           0  2015-12-27          1.33      64236.62  1036.74   54454.85   
1           1  2015-12-20          1.35      54876.98   674.28   44638.81   
2           2  2015-12-13          0.93     118220.22   794.70  109149.67   
3           3  2015-12-06          1.08      78992.15  1132.00   71976.41   
4           4  2015-11-29          1.28      51039.60   941.48   43838.39   

     4770  Total Bags  Small Bags  Large Bags  XLarge Bags          type  \
0   48.16     8696.87     8603.62       93.25          0.0  conventional   
1   58.33     9505.56     9408.07       97.49          0.0  conventional   
2  130.50     8145.35     8042.21      103.14          0.0  conventional   
3   72.58     5811.16     5677.40      133.76          0.0  conventional   
4   75.78     6183.95     5986.26      197.69          0.0  conventional   

   year  region  
0  2015  Albany  
1  2015  Albany  
2  2015  Albany  
3  2015  Albany  
4  2015  Albany

In [ ]:
avocado_df.info()

In [ ]:
# Writing out files - pandas allow you to output dataframes as files as well
new_df = avocado_df.iloc[1:5, 2:7] # guess what this line is doing

new_df.to_csv("some_data.csv")

Now that we understand how to read in data, we will see how loops and functions help to make data analysis easier

Following are the commands necessary for you to carry out the upcoming tasks

In [ ]:
# Getting summary statistics
avocado_df["AveragePrice"].mean() 
avocado_df["Total Volume"].median()
avocado_df["Total Volume"].std()

In [6]:
# Getting column names
names = list(avocado_df)
names

['Unnamed: 0',
 'Date',
 'AveragePrice',
 'Total Volume',
 '4046',
 '4225',
 '4770',
 'Total Bags',
 'Small Bags',
 'Large Bags',
 'XLarge Bags',
 'type',
 'year',
 'region']

In [ ]:
# why is this useful? 
avocado_df[names[2]]

In [7]:
# casting: converting values into strings
str(3.14)

'3.14'

In [7]:
# printing an empty line
print('\n')
# concatenating print statements 
print("a" + "b " + str(123) + names[0])



ab123Unnamed: 0


### Exercise
1. Using a for-loop and if-else statement, how many shipment of avocado have an average price over 1.35?
2. Write a function that takes in a column, and print out the mean, median and standard deviation of the column. Then apply a for-loop to on the columns "Total Bags", "Small Bags" and "Large Bags".